### PR TITLE
Installer checkpointing

### PR DESCRIPTION
**Pull request by mtnbikenc**
This pull request adds an Ansible callback plugin to allow for 'checkpointing' the status of the OpenShift-Ansible installer.  The installer is divided up into phases and each phase will start with a status of 'Not Started'.  As the phase is performed the status will be updated to 'In Progress' and finally 'Complete'.  Should the install fail or complete successfully, the callback plugin will report the status of all phases after the PLAY RECAP.  The INSTALLER STATUS is also written to the Ansible log file.

https://trello.com/c/VNEvT2Fo

Changes:
- Added installer_checkpoint role and callback plugin
- Added checkpoint 'Start' and 'End' plays to each installation phase

Additional items related to enabling proper checkpointing:
- Removed openshift_set_hostname and related task (related to 3.0)
- Added openshift-hosted entry point playbook
- Moved openshift metrics and logging out of openshift_hosted playbook
- Moved API Aggregation play to master install

Example playbook output:
```
<- snip ->
PLAY RECAP ***********************************************************************************************************************************
ec2-34-207-180-218.compute-1.amazonaws.com : ok=158  changed=16   unreachable=0    failed=0   
ec2-52-87-169-12.compute-1.amazonaws.com : ok=469  changed=74   unreachable=0    failed=0   
ec2-52-90-209-20.compute-1.amazonaws.com : ok=157  changed=17   unreachable=0    failed=0   
localhost                  : ok=24   changed=0    unreachable=0    failed=0   


INSTALLER STATUS *****************************************************************************************************************************
Initialization             : Complete
etcd Install               : Complete
NFS Install                : Not Started
Load balancer Install      : Not Started
Master Install             : Complete
Master Additional Install  : Complete
Node Install               : Complete
GlusterFS Install          : Not Started
Hosted Install             : Complete
Metrics Install            : Not Started
Logging Install            : Not Started
Service Catalog Install    : Not Started
```

### CODE REVIEW

File: roles/installer_checkpoint/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+- name: Initialize installer phase status
+  set_stats:
+    data:
```
Comment (kwoodson) [inline roles/installer_checkpoint/tasks/main.yml]: Should this data be set in the `defaults/main.yml` and therefore be passed into the `installer_checkpoint` if we want to set the installer check points manually?

```
data:
      installer_phase_initialize: "Completed"
      installer_phase_etcd: "Retry"
      installer_phase_nfs: "Completed"
      installer_phase_loadbalancer: "Not Started"
      installer_phase_master: "Not Started"
      installer_phase_node: "Not Started"
```

Thoughts?

File: playbooks/common/openshift-master/config.yml [hunk 2]
```diff
@@ -231,10 +242,27 @@
     group_by: key=oo_masters_deployment_type_{{ openshift.common.deployment_type }}
     changed_when: False
 
+- name: Configure API Aggregation on masters
```
Comment (ewolinetz) [inline playbooks/common/openshift-master/config.yml]: 👍

File: playbooks/common/openshift-master/wire_aggregator.yml [hunk 3]
```diff
@@ -74,7 +74,7 @@
   stat:
     path: /etc/origin/master/aggregator-front-proxy.kubeconfig
   register: first_front_proxy_kubeconfig
-  delegate_to: "{{ first_master }}"
+  delegate_to: "{{ groups.oo_first_master.0 }}"
```
Comment (ewolinetz) [inline playbooks/common/openshift-master/wire_aggregator.yml]: :+1:

**Comment by mtnbikenc** [comment]
@tbielawa @kwoodson PTAL
I've got this code up and tested so it is ready for a final review.  An attempt at documenting it has been added to the README.md.

**Comment by mtnbikenc** [comment]
@sdodson If you have a chance I'd like your feedback on this as well to make sure this is headed in the right direction.

**Comment by sdodson** [comment]
I think this is good. I hadn't thought about the optional nature which may lead to scenarios where it's not a clearly linear progression of well defined steps. Maybe for those where we skip an entire phase (nfs) we set a status of `skipped` or something?

I'm fine with this as is though.

**Pull request merged**